frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,471 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { merge, mergeWith, isArray, assign } from "lodash"
+
+import { hasLightBackgroundColor, Theme } from "src/theme"
+
+export function applyStreamlitThemeData(data: any, theme: Theme): any {
+  // This theming config contains multiple hard coded spacing values.
+  // The reason is that we currently only have rem values in our spacing
+  // definitions and vega lite requires numerical (pixel) values.
+  const { genericFonts, colors, fontSizes, fontWeights } = theme
+  const streamlitTheme = {
+    autocolorscale: true,
+  }
+
+  if (!data) {
+    return streamlitTheme
+  }
+
+  return assign(data, streamlitTheme)
+}
+
+function getGreyText(theme: Theme) {
```

Comment:
I don't think you actually need this, using `theme.colors.fadedText60` everywhere instead should be enough. Atleast that is how the altair theme does it. The color of `theme.colors.fadedText60` is automatically changed depending on the theme.

Reply:
Hm I was using `fadedText70` and I just tested it out but it didn't seem to work properly. Arnaud specifically said that he didn't want that color of `fadedText70`.
<img width="821" alt="Screen Shot 2022-10-07 at 10 31 32 AM" src="https://user-images.githubusercontent.com/16749069/194616172-69537447-9479-42e1-9840-5a3a94d59173.png">

Reply:
Hmm, did you had `fadedText70` or `fadedText60`

Reply:
Hm I recently tested both, but I did have fadedText60 because I was trying to mimic properties in st.altair chart

Reply:
Maybe we need to clarify if we also should change this in the altair theme. I think having two different font color setups between plotly & altair probably does not make a lot of sense.